File: everfrost/Agent_of_Change.lua
local aoc_raid = {
    expedition = { name="Lair of Lady Vox", min_players=1, max_players=72},
    instance   = { zone="permafrost", version=0, duration=eq.seconds("8h") }, -- zone lasts 8 hours
    safereturn = { zone="everfrost", x=-7010, y=2033, z=-60.1, h=315.25 },
    zonein     = { x=61, y=-121, z=2, h=256 },
}

function event_say(e)
	local dz = e.other:GetExpedition()
	if(e.message:findi("hail")) then
		if dz.valid and dz:GetZoneName() == aoc_raid.instance.zone then
			e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
		else
			e.self:Say("Would you like to [" .. eq.say_link("request") .. "] the expedition?")
		end
	elseif(e.message:findi("request")) then
		local dz = e.other:CreateExpedition(aoc_raid)
		dz:AddReplayLockout(eq.seconds("9h")) -- 9 hour lockout
		e.self:Say("Tell me when you're [" .. eq.say_link("ready") .. "] to enter")
	elseif(dz:GetZoneName() == aoc_raid.instance.zone and e.message:findi("ready")) then
		e.other:MovePCDynamicZone(aoc_raid.instance.zone)
	end
end

File: everfrost/Agent_of_Change.pl
#agent of change by osirous

my $expedition_name = "Lair of Lady Vox";
my $min_players     = 1;
my $max_players     = 40;
my $dz_zone         = "permafrost";
my $dz_version      = 0;
my $dz_duration     = 28800; # 8 hours | #3600; # 1 hour

sub EVENT_SAY {
  if ($text =~ /hail/i) {
    my $dz = $client->GetExpedition();
    if ($dz && $dz->GetName() eq $expedition_name) {
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
    else {
      quest::say("Would you like to [" . quest::saylink("request") . "] the expedition?");
    }
  }
  elsif ($text =~ /request/i) {
    my $dz = $client->CreateExpedition($dz_zone, $dz_version, $dz_duration, $expedition_name, $min_players, $max_players);
    if ($dz) {
      $dz->SetSafeReturn("everfrost", -7010.22, 2033.74, -60.72, 315.25); # outside portal
      $dz->SetZoneInLocation(60.93, -123.97, 3.13, 258.25); # normal zone in place
      $dz->AddReplayLockout(32400); # immediately add a 9 hour replay lockout on creation
      quest::say("Tell me when you're [" . quest::saylink("ready") . "] to enter");
    }
  }
  elsif ($text =~ /ready/i) {
    $client->MovePCDynamicZone("permafrost");
  }
}